Slim/Utils/Scanner/Remote.pm

```diff
@@ -842,7 +842,9 @@ sub parseMp4Header {
 	Slim::Music::Info::setDuration( $track, $duration );
 	
 	# use the audio block to stash the temp file handler
-	$track->initial_block_fh($info->{fh});
+	$track->initial_block_fn($info->{fh}->filename);
+	$info->{fh}->unlink_on_destroy(0);
```

Comment:
My question is coming a bit late... but why not unlink the file? I indeed see a file in`/tmp` for every lossy TIDAL track I play...

Reply:
Because I need the header for every seeking in the $song (for some formats). When the $song is not modified, it's always the same $track element that is being used. I think (but I've not verified) that as long as the $track is in cache, LMS will use it direcly for another $song item as well. 

For flac, the header does not need to be kept so the file is removed when closed and it is closed when it goes out of scope. 

Now, when the $track cache is full, the LRU cache will do its job and destroy the least used one and at that time I'll unlink the file (it also happens when LMS stops, you can verify that /tmp is cleanup when LMS exist or restarts)